385 enhancement help documentation for gmp module

aaron-auc opened the pull request:
When checking this pull request take a close look at the Make file, manual.qhp, and readme. Most of the changes deal with organizing files and images in directories. I ran linkchecker and only the template file showed a broken link and not any of the help files. Please verify that this help works properly in UltraScan after completing a proper build and within the UltraScan GUI. My testing has focused on local builds and reviewed in QT Assistant.

demeler reviewed (commented):
I wasn't able to build the manual on either my desktop or uslims.uleth.ca. Maybe Emre can help?
Here is the error I got:

Generating Qt help files...
/bin/sh: line 1: qhelpgenerator: command not found
make: *** [Makefile:36: ../../bin/manual.qch] Error 127

ehb54 commented:
> @aaron-auc: I wasn't able to build the manual on either my desktop or uslims.uleth.ca. Maybe Emre can help - @ehb54? Here is the error I got:
> 
> Generating Qt help files... /bin/sh: line 1: qhelpgenerator: command not found make: *** [Makefile:36: ../../bin/manual.qch] Error 127
> 
> Looks like an incomplete Qt install.

Yep, QDoc wasn't created & assume that's why qthelpgenerator wasn't made.
```
WARNING: QDoc will not be compiled, probably because libclang could not be located. This means that you cannot build the Qt documentation.

Either ensure that llvm-config is in your PATH environment variable, or set LLVM_INSTALL_DIR to the location of your llvm installation.
On Linux systems, you may be able to install libclang by installing the libclang-dev or libclang-devel package, depending on your distribution.
On macOS, you can use Homebrew's llvm package.
On Windows, you must set LLVM_INSTALL_DIR to the installation path.
```

So we will have to make sure llvm & clang are installed and rebuild qt :(

ehb54 commented:
> you can use the apt package `sudo apt install qhelpgenerator-qt5` that worked for me quite well. Checking now the generated files

Hmm. perhaps there's a dnf version also. Might be a workaround. Ideally should be part of our build process & not sure if installing that is better than bringing in the components needed to properly build. Also might be a problem if the repos drop qt5 in future.

doluk commented:
still not centered images:
 - us_edit.html
 - load/load_auc.html
 - us_edit_mwl.html
 - uscolor2.html
 - config2.html
 - vhw_enhanced.html
 - grid_editor.html
 
 layout:
 - config.html has the second image next to text. on small windows this gets hard to read
 - strange spacing at the top vhw_distrib_plot.html
 
 dead links:
 - vhw_enhanced.html link to vHW Groups Image is dead (vhw_enh_groups.img.html)
 
 
 will continue later (have pick up the child)

aaron-auc commented:
@doluk I updated to try to address centering.  I couldn't replicate the dead links in my environment. Layout is something you can create a new issue for unless you believe that the layout is a bug that I introduced.

doluk commented:
Might be due to some mixture of wsl build and windows. I will try it again on a pure linux machine tomorrow

doluk commented:
I built everything on pure linux

load_auc.html shows for me still not centered images
![image](https://github.com/user-attachments/assets/dba1c5d6-e7a3-4526-ad20-df3d74d3383f)
qthelp://ultrascaniii/manual/vhw_enh_groups.img.html shows still empty for me (that is the vHw groups link from vhw-enhanced)

aaron-auc commented:
@doluk Are you using the latest from my branch? 
![image](https://github.com/user-attachments/assets/a4df03a3-7101-4a9c-8d80-8be67ebcc33b)

And I'm getting the image. I am using make on a MacOS and viewing it through QT Assistant. I can see a difference in rendering, but a missing page is perplexing. 
![image](https://github.com/user-attachments/assets/6ebc54d8-fa89-4c12-8d37-a4194030f3df)

During the build it should create a  doc/manual/.staging directory and in that there should be a vhw_enh_groups.img.html, can you verify that is being created in your environment? If not, you should be able to run make in doc/manual for it to create the directory and the compressed help that the QT Help Framework uses.

aaron-auc commented:
I don't know. I deleted everything from bin and the directory manual and all subdirectories. I then copied the archive manual that you had sent. I ran make clean and then make. Linked worked to qthelp://ultrascaniii/manual/vhw_enh_groups.img.html.

I appreciate the steps you sent. I followed the directions by cutting and pasting each command which put everything into a new directory us3-temp. The link for qthelp://ultrascaniii/manual/vhw_enh_groups.img.html worked in assistant from the vmh enhanced page. I also tried shutting down assistant, opening with no help associated, and then restarting with ultrascan help thinking maybe something was cached. Still, everything worked fine. 

Like you, I was thinking that something was leftover in my environment causing the issue, but after starting over and deleting directories and trying a new directory, I don't see how that could be.  

When you have a new directory and run make, you do see vhw_enh_groups.img.html being logged as created, don't you?

doluk commented:
Yes, it is listed in the output, see below
```
Preparing staging directory...
2dsa_3dplot.png  WRAPPED TO:  .staging/2dsa_3dplot.img.html
2dsa_resplot.png  WRAPPED TO:  .staging/2dsa_resplot.img.html
analyte_details.png  WRAPPED TO:  .staging/analyte_details.img.html
buffer_info.png  WRAPPED TO:  .staging/buffer_info.img.html
cfa_view-loadraw.png  WRAPPED TO:  .staging/cfa_view-loadraw.img.html
cfa_view-loadauc.png  WRAPPED TO:  .staging/cfa_view-loadauc.img.html
cfa_view-statsraw.png  WRAPPED TO:  .staging/cfa_view-statsraw.img.html
cfa_view-statsauc.png  WRAPPED TO:  .staging/cfa_view-statsauc.img.html
ddist_combine-selrun.png  WRAPPED TO:  .staging/ddist_combine-selrun.img.html
fit_men-editupd.png  WRAPPED TO:  .staging/fit_men-editupd.img.html
fit_men-load.png  WRAPPED TO:  .staging/fit_men-load.img.html
grid_editor.png  WRAPPED TO:  .staging/grid_editor.img.html
grid_editor_multi_grids.png  WRAPPED TO:  .staging/grid_editor_multi_grids.img.html
grid_editor_multi_grids_sed.png  WRAPPED TO:  .staging/grid_editor_multi_grids_sed.img.html
grid_editor_mw_view.png  WRAPPED TO:  .staging/grid_editor_mw_view.img.html
mwlr_view_3dplot.png  WRAPPED TO:  .staging/mwlr_view_3dplot.img.html
pcsa_3dplot.png  WRAPPED TO:  .staging/pcsa_3dplot.img.html
pcsa_mlines-line1.png  WRAPPED TO:  .staging/pcsa_mlines-line1.img.html
pcsa_mlines-line2.png  WRAPPED TO:  .staging/pcsa_mlines-line2.img.html
pcsa_mlines-sigm1.png  WRAPPED TO:  .staging/pcsa_mlines-sigm1.img.html
pcsa_mlines-sigm2.png  WRAPPED TO:  .staging/pcsa_mlines-sigm2.img.html
pcsa_resplot.png  WRAPPED TO:  .staging/pcsa_resplot.img.html
reporter-details.png  WRAPPED TO:  .staging/reporter-details.img.html
reporter-loadprof.png  WRAPPED TO:  .staging/reporter-loadprof.img.html
reporter-save.png  WRAPPED TO:  .staging/reporter-save.img.html
reporter-saveprof.png  WRAPPED TO:  .staging/reporter-saveprof.img.html
reporter-view.png  WRAPPED TO:  .staging/reporter-view.img.html
reporter-viewitem.png  WRAPPED TO:  .staging/reporter-viewitem.img.html
sequence.png  WRAPPED TO:  .staging/sequence.img.html
us_buffer_spect.png  WRAPPED TO:  .staging/us_buffer_spect.img.html
vhw_combine-selrun.png  WRAPPED TO:  .staging/vhw_combine-selrun.img.html
vhw_enh_groups.png  WRAPPED TO:  .staging/vhw_enh_groups.img.html
Processing HTML files...
Generating Qt help files...
Building up file structure...
Insert custom filters...
Insert help data for filter section (1 of 1)...
Insert files...
Insert contents...
Insert indices...
Documentation successfully generated.
Reading collection config file...
Generating help for manual.qhp...
Building up file structure...
Insert custom filters...
Insert help data for filter section (1 of 1)...
Insert files...
Insert contents...
Insert indices...
Documentation successfully generated.
Creating collection file...
Installing Qt help files to ../../bin...
Generating Qt help files...
Building up file structure...
Insert custom filters...
Insert help data for filter section (1 of 1)...
Insert files...
Insert contents...
Insert indices...
Documentation successfully generated.
Reading collection config file...
Generating help for manual.qhp...
Building up file structure...
Insert custom filters...
Insert help data for filter section (1 of 1)...
Insert files...
Insert contents...
Insert indices...
Documentation successfully generated.
Creating collection file...
Installing Qt help files to ../../bin...
```
the content of `.staging/vhw_enh_groups.img.html` looks fine to me. nothing unusual or unexpected.
opening it in the qtassistant from the command line i just saw the following message `QTextBrowser: No document for qthelp://ultrascaniii/manual/vhw_enh_groups.img.html`. So maybe the path is wrong? Could I miss some sort of dependency which could explain the behavior I observe?

aaron-auc commented:
Can you send me the files in bin? The manual.qch and manual.qhc?

doluk commented:
They are part of the archive I shared with you earlier

aaron-auc commented:
I'm out of ideas. I started QT Assistant with no help associated. Then added the qch file from the archive. Navigated to Velocity Data Analysis -> Enhanced van Holde - Weishet -> click on link vHW Groups image and it loaded without error.

doluk commented:
Could this be an issue with me missing any sort of packages? The documentation lists a certain cpan package? I tried to install that on windows/wsl but not on the pure linux machine.

aaron-auc commented:
Do other wrapped html files appear correctly? For instance, does 2dsa_3dplot.img.html render correctly?

doluk commented:
No they all fail to be displayed.

But the odd thing is, if you look into the generated qhc and qch files (they are just sqlite databases), you can see the wrapped images in there. And they have some binary content, which I wasn't able to decode properly.

I found a fix to the not showing wrapped image. By running `assistant -register manual.qch` before then running `assistant` it works. `assistant -collectionFile manual.qhc` doesn't work.

@aaron-auc No idea how to go from here, especially because it just worked magically for you.

demeler commented:
Building on demeler9, I still get failures (on 195, using us_branch.php):

Processing HTML files...
Generating Qt help files...
/bin/sh: line 1: qhelpgenerator: command not found
make: *** [Makefile:36: ../../bin/manual.qch] Error 127

On Thu, Apr 10, 2025 at 1:28 PM ehb54 ***@***.***> wrote:

> @ehb54 <https://github.com/ehb54> requested your review on: #195
> <https://github.com/ehb54/ultrascan3/pull/195> 385 enhancement help
> documentation for gmp module.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ehb54/ultrascan3/pull/195#event-17212384976>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2N726LQSIO2HY3THBZSP32Y3A7FAVCNFSM6AAAAAB2BYVNDSVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGIYTEMZYGQ4TONQ>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

ehb54 commented:
@demeler 
Only upgraded qt (with qhelpgenerator etc) on U Montana.
Can upgrade demeler9, but need a time slot (an hour or so) to get it also upgraded.

demeler commented:
The main GMP Data Acquisition window "Help" button doesn't seem to have a help slot attached to it. Ditto for e-signatures and protocol development. I also missed helpfiles in for various other functions in us_analyte. Someone needs to go through UltraScan and write those missing files, and connect the appropriate slots to the help buttonclicked() signal.

aaron-auc commented:
@demeler I will see what I can do about connecting the "Help" button to current help files.

demeler reviewed (approved):
Documentation improvements are helpful for our current release. More work can be done later.
